04.ls/wc.rb

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  cmd_arg = receive_command_argument
+  opt_params = cmd_arg[0]
+  argv = cmd_arg[1]
+  if argv.empty?
+    std_data = make_std
+    output(std_data, opt_params)
+  else
+    lwc_data = make_lwc(argv)
+    output(lwc_data, opt_params)
+  end
+end
+
+def receive_command_argument
+  opt_params = { l: false, w: false, c: false }
+  opt = OptionParser.new
+
+  opt.on('-l') { |v| opt_params[:l] = v }
+  opt.on('-w') { |v| opt_params[:w] = v }
+  opt.on('-c') { |v| opt_params[:c] = v }
+  argv = opt.parse(ARGV)
+  [opt_params, argv]
+end
+
+def make_std
+  inputs = $stdin.readlines
+  formatted_inputs = inputs.join

```

Comment:
```suggestion
  text = $stdin.readlines.join
```

みたいにした方が簡潔です
（変数名のformattedというのもこの処理では特に整形らしい整形処理はないのであまり適切ではないですね）

Reply:
修正いたしました。